src/ncp/ncp_base_mtd.cpp

```diff
@@ -3744,13 +3744,13 @@ void NcpBase::ProcessThreadChangedFlags(void)
 
     // Convert OT_CHANGED flags to corresponding NCP property update.
 
-    for (size_t i = 0; i < OT_ARRAY_LENGTH(kFlags); i++)
+    for (auto kFlag : kFlags)
```

Comment:
Use `flag` instead of `kFlag`?
```suggestion
    for (auto flag : kFlags)
```